var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -986,6 +990,16 @@ def cmake_args(self):
 
         return cmake_args
 
+    @run_after("cmake")
+    def change_makefile(self):
+        # Avoid Fujitsu compiler Clang Mode options when building LLVM
+        if self.spec.satisfies("%fj"):
+            filter_file(
+                r"-DCMAKE_C_COMPILER=",
+                "-DCMAKE_CXX_FLAGS= -DCMAKE_C_COMPILER=",
+                join_path(self.build_directory, "build.ninja"),
+            )
```

Comment:
This is really concerning to me. 
* This should use `when=` in `run_after` to limit to running with `%fj` rather than checking in the function.
* The function name implies it modifies a makefile, it modifies a ninja file.
* The modification injects an empty `CMAKE_CXX_FLAGS` definition anywhere the `CMAKE_C_COMPILER` definition exists, which should matter mainly for invocations of cmake, _after_ cmake has already configured the build.  It also may either have no effect at all, since it could be overridden, or clears all C++ compiler flags specified to cmake. 

The name and when parts are easy to fix, but the latter makes me wonder what the intended outcome is, and if this reliably accomplishes it.

Reply:
@trws 
Thank you for your review. I thought your third point was reasonable.
I took the time to check and see that the workaround is to patch the llvm source instead of putting a fix in `package.py`.
Therefore, I have removed the fix you pointed and put the fix in `llvm17-fujitsu.patch`([L15-L28](https://github.com/spack/spack/pull/43387/files#diff-8cefbc6041883afe5d0d593ef4804a03d870584b9f8d0ef39c0cb7d3bca7f27bR15-R28)).